jscomp/core/js_of_lam_block.ml

```diff
@@ -57,11 +57,12 @@ let field (field_info : Lam_compat.field_dbg_info) e i =
   | Fld_record_inline comment
   | Fld_record_extension comment
 #end
-  | Fld_record comment
-    -> E.array_index_by_int ~comment e i
+  | Fld_record comment ->
+    E.dot e comment
```

Comment:
can we do something similiar as `module_access` below to preserve as much as semantics as possible